resolver_test.go

```diff
@@ -35,3 +35,70 @@ func TestExhaustiveErrors(t *testing.T) {
 
 	assert.JSONEq(t, `{"title":"Bad Request","status":400,"detail":"Error while parsing input parameters","errors":[{"message":"cannot parse boolean","location":"query.bool","value":"bad"},{"message":"cannot parse integer","location":"query.int","value":"bad"},{"message":"cannot parse float","location":"query.float32","value":"bad"},{"message":"cannot parse float","location":"query.float64","value":"bad"},{"message":"cannot parse integer","location":"query[2].tags","value":"bad"},{"message":"unable to validate against schema: invalid character 'b' looking for beginning of value","location":"query.tags","value":"[1,2,bad]"},{"message":"cannot parse time","location":"query.time","value":"bad"},{"message":"Must be greater than or equal to 5","location":"body.value","value":1}]}`, w.Body.String())
 }
+
+type Dep1 struct {
+	Value    string `json:"value"`
+	Computed string `json:"-"`
+}
+
+func (d *Dep1) Resolve(ctx Context, r *http.Request) {
+	if d.Value == "error" {
+		ctx.AddError(&ErrorDetail{
+			Message:  "msg",
+			Location: "value",
+			Value:    d.Value,
+		})
+	}
+
+	d.Computed = strings.ToUpper(d.Value)
+}
+
+type Dep2 struct {
+	Foo map[string][]Dep1 `json:"foo"`
+}
+
+func TestNestedResolver(t *testing.T) {
+	app := newTestRouter()
+
+	app.Resource("/").Post("test", "Test",
+		NewResponse(http.StatusOK, "desc").ContentType("text/plain"),
+	).Run(func(ctx Context, input struct {
+		Body Dep2
+	}) {
+		computed := []string{}
+
+		for _, v := range input.Body.Foo {
+			for _, dep1 := range v {
+				computed = append(computed, dep1.Computed)
+			}
+		}
+
+		ctx.WriteHeader(http.StatusOK)
+		ctx.Write([]byte(strings.Join(computed, ", ")))
+	})
+
+	// Test happy case
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest(http.MethodPost, "/", strings.NewReader(`{"foo": {"one": [{"value": "v1"}], "two": [{"value": "v2"}]}}`))
+	app.ServeHTTP(w, r)
+
+	assert.Equal(t, "V1, V2", w.Body.String())
+
+	// Test error case
+	w = httptest.NewRecorder()
+	r, _ = http.NewRequest(http.MethodPost, "/", strings.NewReader(`{"foo": {"one": [{"value": "error"}]}}`))
```

Comment:
It took me a bit to parse out what this test is doing because the JSON is valid, so I had to stop and work out how the error was being injected. Is there a way to test this without injecting an artificial failure? What would normally trigger a failure case in a real-world code example?

Reply:
Good point. I've updated this for my real use-case: an exclusive or (one-of) scenario. Take another look and let me know what you think!